linear_algebra/src/lib.py

```diff
@@ -183,6 +182,16 @@ def angle(self, other: Vector, deg: bool = False) -> float:
         else:
             return math.acos(num / den)
 
+    def __eq__(self, vector: object) -> bool:
+        """
+        performs the comparison between two vectors
+        """
+        if not isinstance(vector, Vector):
+            return NotImplemented
```

Comment:
```suggestion
    def __eq__(self, vector: Vector) -> bool:
        """
        performs the comparison between two vectors
        """
```
I don't think it makes sense to allow comparison with any object, only with other `Vector`s.

Reply:
I tried writing Vector insted of object but it gave me an error that saying: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]
This violates the Liskov substitution principle